### PR TITLE
Task-44573: Add the maximum number of characters counter for task description

### PR DESCRIPTION
**Pull request by wassimmhamdi**
Problem: The task description field has no indication on the description's maximum number of characters
Fix: we add the character counter to make a limitation of the task's description which is 2000 characters like the one of project description field, so we must make the add buttom disabled whem the title of the task is less thean 3 caracters or more than 1024 and the max of descreption is more than 2000 caracter

### CODE REVIEW

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue [hunk 1]
```diff
@@ -366,6 +374,13 @@ export default {
     document.removeEventListener('keyup', this.escapeKeyListener);
   },
   methods: {
+    loadDescription(description) {
+      let pureText = description ? description.replace(/<[^>]*>/g, '').replace(/&nbsp;/g, '').trim() : '';
+      const div = document.createElement('div');
+      div.innerHTML = pureText;
+      pureText = div.textContent || div.innerText || '';
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue:381]: - The param sent by the emited event has already treated by this same code, why doing the same treatment again ?
- You can directly the param isDescriptionValid

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue [hunk 2]
```diff
@@ -89,9 +88,20 @@ export default {
...
+      val = pureText;
+      this.$root.$emit('max', val);
     },
-    value() {
+
+    value(val) {
       this.inputVal = this.taskDescription;
+      if (val >= 2000) {
+        this.$root.$emit('max');
+      }
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue:104]: do you need to emit the event also from this watch ?

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue [hunk 3]
```diff
@@ -89,9 +88,20 @@ export default {
           CKEDITOR.instances['descriptionContent'].setData(val);
         }
       }
+      // Modify input Val from HTML to text and send it in event
+      let pureText = val ? val.replace(/<[^>]*>/g, '').replace(/&nbsp;/g, '').trim() : '';
+      const div = document.createElement('div');
+      div.innerHTML = pureText;
+      pureText = div.textContent || div.innerText || '';
+      val = pureText;
+      this.$root.$emit('max', val);
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue:97]: you can directly send as event parameter if the description is valid or not and use it directly when intercepting the emited event in TaskDrawer.vue

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue [hunk 4]
```diff
@@ -366,6 +374,13 @@ export default {
     document.removeEventListener('keyup', this.escapeKeyListener);
   },
   methods: {
+    loadDescription(description) {
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue:377]:
```suggestion
    isDescriptionValid(descriptionValid) {
```

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue [hunk 5]
```diff
@@ -342,6 +347,9 @@ export default {
       }
     });
 
+
+    this.$root.$on('max', event => this.loadDescription(event));
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue:351]:
```suggestion
    this.$root.$on('max', descriptionValid=> this.descriptionValid(event));
```

**Review by azayati — CHANGES_REQUESTED** [review]
see review

**Comment by wassimmhamdi** [comment]
closed for new commit and review

**Pull request closed without merging**